plugins/flytekit-neptune/flytekitplugins/neptune/tracking.py

```diff
@@ -0,0 +1,106 @@
+import os
+from functools import partial
+from typing import Callable, Union
+
+import neptune
+from flytekit import Secret
+from flytekit.core.context_manager import FlyteContext, FlyteContextManager
+from flytekit.core.utils import ClassDecorator
+
+NEPTUNE_RUN_VALUE = "neptune-run-id"
+
+
+def neptune_init_run(
+    project: str,
+    secret: Union[Secret, Callable],
+    host: str = "https://app.neptune.ai",
```

Comment:
Is there any particular reason `host` has to be passed explicitly?
This can be extracted from the `NEPTUNE_API_TOKEN`

```json.loads(base64.b64decode(os.getenv("NEPTUNE_API_TOKEN")).decode("utf-8"))["api_url"]```

Reply:
In a IRL conversation with @SiddhantSadangi, we discussed how this is a limitation of Flyte and dynamic log links.

Context: The Flyte UI needs to know the host URL during registration time. There are two options:

1. Have the host passed in explicitly (this PR)
2. Force users to have `NEPTUNE_API_TOKEN` defined during registration time.